src/python/pants/backend/python/util_rules/pex.py

```diff
@@ -157,6 +158,7 @@ def __init__(
         *,
         output_filename: str,
         internal_only: bool,
+        python: PythonExecutable | None = None,
```

Comment:
This is actually super useful if we end up reviving https://github.com/pantsbuild/pants/pull/12389. Thanks!